### DISCUSSION TITLE
Setup Teleport Site on kubernetes but unable to install the ssh agent to connect to the node

### DISCUSSION THREAD

**Discussion by unixbird** [gravitational/teleport#46018] [Q&A] [answered]
I have a pretty simple setup and have teleport running in my 1 node kubernetes cluster. I got it setup and working with another Linux host of mine and it works perfectly but once I attempt to run the installer for this host (that the teleport container sits on) it says that the teleport process is running and it cannot proceed. I found after killing the process a couple of times I realized that its detecting the container process which is stopping the install from continuing.

I found the same scenario that's happening but no real answer: https://github.com/gravitational/teleport/discussions/14951

Is there a way to get around this?

**Comment by webvictim** [comment, marked as answer]
The script is intentionally naive and just checks whether there's already a `teleport` process running on the host - it's seeing the `teleport` process running inside Docker and thinks it's already installed/running.

For more advanced use cases like this, you can work around the script's limitations by manually installing Teleport on the host, then generating your own config file and joining it to the cluster as described in the docs: https://goteleport.com/docs/enroll-resources/server-access/getting-started/#step-14-install-teleport-on-your-linux-host

**Reply by unixbird** [reply]
I was actually working on this when I figured out that the systemd unit file just does the whole "teleport start" with the yaml config instead of the script doing something magical :P Ended up going through the instructions in your link and just copying the systemd unit file from a working teleport agent I had and it works now. Thank you!
